tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/jms/multiprotocol/JMSMessageConsumerTest.java

```diff
@@ -193,4 +195,66 @@ private void testDurableSubscriptionWithConfigurationManagedQueue(ConnectionSupp
          assertEquals("color = 'BLUE'", queue.getFilter().getFilterString().toString());
       }
    }
+
+   @Test(timeout = 30000)
+   public void testMapMessageConversionBetweenOpenWireAndAMQP() throws Exception {
+      testMapMessageConversion(createOpenWireConnection(), createConnection());
+   }
+
+   @Test(timeout = 30000)
+   public void testMapMessageConversionBetweenAMQPAndOpenWire() throws Exception {
+      testMapMessageConversion(createConnection(), createOpenWireConnection());
+   }
+
+   @Test(timeout = 30000)
+   public void testMapMessageConversionBetweenCoreAndAMQP() throws Exception {
+      testMapMessageConversion(createCoreConnection(), createConnection());
+   }
+
+   @Test(timeout = 30000)
+   public void testMapMessageConversionBetweenAMQPAndCore() throws Exception {
+      testMapMessageConversion(createConnection(), createCoreConnection());
+   }
+
+   @Test(timeout = 30000)
+   public void testMapMessageConversionBetweenCoreAndOpenWire() throws Exception {
+      testMapMessageConversion(createCoreConnection(), createOpenWireConnection());
+   }
+
+   @Test(timeout = 30000)
+   public void testMapMessageConversionBetweenOpenWireAndCore() throws Exception {
+      testMapMessageConversion(createOpenWireConnection(), createCoreConnection());
+   }
+
+   private void testMapMessageConversion(Connection sender, Connection consumer) throws Exception {
+      final String NAME = "myPropertyName";
+      final String VALUE = RandomUtil.randomString();
+
+      try {
+         Session session1 = sender.createSession(false, Session.AUTO_ACKNOWLEDGE);
+         Session session2 = consumer.createSession(false, Session.AUTO_ACKNOWLEDGE);
+
+         javax.jms.Queue queue1 = session1.createQueue(getQueueName());
+         javax.jms.Queue queue2 = session2.createQueue(getQueueName());
+
+         final MessageConsumer consumer2 = session2.createConsumer(queue2);
+
+         MessageProducer producer = session1.createProducer(queue1);
+         producer.setDeliveryMode(DeliveryMode.PERSISTENT);
+         sender.start();
+
+         MapMessage message = session1.createMapMessage();
+         message.setStringProperty(NAME, VALUE);
+         producer.send(message);
+
+         Message received = consumer2.receive(100);
```

Comment:
100 seems a little low for a message we do expect to receive, could be unreliable in slow CI envs.